sdk/core/core-http/lib/policies/msRestUserAgentPolicy.browser.ts

```diff
@@ -9,7 +9,7 @@
 import { TelemetryInfo } from "./userAgentPolicy";
 
 interface NavigatorEx extends Navigator {
-  readonly oscpu: string | undefined;
+  readonly oscpu: string;
```

Comment:
Ok, reverted this change for now since merge from master will take care of it.

(Unable to respond to some other comments, is it because they are resolved?)

Overall changes:
- Simplified the parseResponse implementation and added better docs. 
- Using the `isKnownResponseCode` still doesn't help with the types; accessing the value still needed the `any` type to be specified.
- Addressed `contentElementName` check
- Changing implementations between sync and async is not trivial, I doubt such a big change would be made as it would be a breaking one. Trying to be extra careful makes sense, but it isn't trivial and can be picked up later when we revisit the `xmlbuilder` dependency in itself.

Reply:
Edit: Given current impl of core-http, it wasn't that difficult - updated to ensure we are awaiting on `parseString`. (Also oops, this was on `xml2js` and not `xmlbuilder`)

Reply:
Interesting that `isKnownResponseCode` wasn't helping, I was playing with it myself and would be curious where you were getting hung up, but at this point I don't think it should be a blocker since you've got the right checks in place.

Reply:
It was still needing the `any` type, I don't recall why - might have been the scoping.
It does work, updated to use this.

Reply:
Marking as resolved as this change is not part of the final set of changes in this PR